src/pylibsshext/logging.pyx

```diff
@@ -0,0 +1,94 @@
+#
+# This file is part of the pylibssh library
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see file LICENSE.rst in this
+# repository.
+
+import contextlib
+import logging
+
+from pylibsshext.errors cimport LibsshSessionException
+from pylibsshext.includes cimport callbacks, libssh
+
+
+ANSIBLE_PYLIBSSH_TRACE = int(logging.DEBUG / 2)
+
+
+LOG_MAP = {
+    logging.NOTSET: libssh.SSH_LOG_TRACE,
+    ANSIBLE_PYLIBSSH_TRACE: libssh.SSH_LOG_TRACE,
+    logging.DEBUG: libssh.SSH_LOG_DEBUG,
+    logging.INFO: libssh.SSH_LOG_INFO,
+    logging.WARNING: libssh.SSH_LOG_WARN,
+    logging.ERROR: libssh.SSH_LOG_WARN,
+    logging.CRITICAL: libssh.SSH_LOG_NONE,
+}
+
+LOG_MAP_REV = {
+    **{
+        libssh_name: py_name
+        for py_name, libssh_name in LOG_MAP.items()
+    },
+}
+
+logger = logging.getLogger("libssh")
+
+
+def _add_trace_log_level():
+    level_num = ANSIBLE_PYLIBSSH_TRACE
+    level_name = "TRACE"
+
+    logging.addLevelName(level_num, level_name)
+
+
+cdef void _pylibssh_log_wrapper(int priority,
+                                const char *function,
+                                const char *buffer,
+                                void *userdata) noexcept nogil:
+    with gil:
+        log_level = LOG_MAP_REV[priority]
+        logger.log(log_level, str(buffer))
+
+
+def _set_log_callback():
+    callbacks.ssh_set_log_callback(_pylibssh_log_wrapper)
+
+
+def _initialize_logging():
+    with contextlib.suppress(AttributeError):
```

Comment:
@Jakuje could you check if this is still needed? It seems that you removed things that might've been able to trigger this exception. And if there's anything else, the suppression should probably live closer to things that would cause it, inside `_add_trace_log_level()`.

Reply:
No needed anymore obviously. Thanks for catching this!